komodo/release_transpiler.py

```diff
@@ -43,28 +72,68 @@ def _build(packages, py_ver, rhel_ver):
             version = versions
 
         if version:
-            release_dict[p] = version
+            release_dict[pkg_name] = version
     return release_dict
 
 
-def transpile_releases(matrix_file, output_folder):
+def _check_consistency_in_versions(
+    versions: dict, pkg_name: str, rhel_ver: str, py_ver: str
+) -> None:
+    """Check the coordinates `rhel_ver` and `py_ver` inputed as arguments to
+    build a release against the release matrix file. Raise exceptions if
+    coordinates not found.
+    """
+    first_level_versions = []
+    for version in versions:
+        first_level_versions.append(version)
```

Comment:
the name `first_level_versions` points to something magical (what is the first level)? I see also that we don't assume which comes first. that's i think too liberal - i think we can pin down the sequence of coordinates, i think that is consistent with the whole "matrix" analogy. so perhaps rename to `rhel_versions`, or if you choose to rename `rhel_ver` to `rhel_coordinate`, then this could be renamed to `rhel_coordinate_values` - or if you wanna move to a more general thing, the arguments could be called `first_coord`, `second_coordinate`, and correspondingly `first_level_versions` could become `first_coordinate_values`.

Reply:
after looking at the input we allow here, it becomes clear that more meaning cannot be added to the names, as we are way too permissive with the inputs we allow (I think). will create a separate issue to look at the input validation.

Reply:
We chatted that the better approach would be to include a longer comment, explaining that `pkg_versions` can take various formats.